### ISSUE TITLE
code for the yaml file

### ISSUE THREAD

**Issue by rmarkovic00** [ultralytics/yolov5#13515]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, I'm trying to pre-train the yolov5m model downloaded from the ultralitics site on my custom dataset which has 3 classes: car,truck person. Is it necessary to match the class ID with the class ID of the coco dataset in yaml file? For example: 

yaml file:
train: /path_to_train_images/
val: /path_to_val_images/

nc: 8  # Broj klasa od 0 do 7 (moraš uključiti sve klase do najvećeg ID-a!)

names: ["Person", "Unknown", "Car", "Unknown", "Unknown", "Unknown", "Unknown", "Truck"]

or its just okej like this:

yaml file:
train: /path_to_train_images/
val: /path_to_val_images/

nc: 8  

names: ["Person", "Unknown", "Car", "Unknown", "Unknown", "Unknown", "Unknown", "Truck"]

because in coco dataset person has ID=0 car=2 and truck=7.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rmarkovic00, thank you for your interest in YOLOv5 🚀 and for your question regarding custom dataset configuration! 

It looks like you're trying to set up your YAML file for training YOLOv5 on a custom dataset. Here's some general guidance:

- For **custom training questions**, it's important to ensure your dataset and YAML configuration are properly set up. The `names` in the YAML file should correspond to the classes in your dataset, and their indices (IDs) should match the order in which they appear in the `names` list, rather than aligning with COCO class IDs. YOLOv5 does not require any alignment with COCO for custom datasets. Please ensure your dataset annotations are consistent with your `names` list.

If this is a 🐛 **Bug Report**, we kindly ask you to provide a **minimum reproducible example (MRE)**, including details on reproducible YAML configurations, relevant code snippets, debugging logs, and any other information to help us diagnose the issue more effectively 🚀.

## Additional Resources

Please also refer to our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) for more guidance:
- Learn all about [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/).
- Avoid common pitfalls with these [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements

Your environment should meet the following:
- [**Python>=3.8.0**](https://www.python.org/) installed with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies, including [**PyTorch>=1.8**](https://pytorch.org/).

To verify installation, run:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Verified Environments

YOLOv5 supports the following environments for training and running models:
- Free GPU-enabled notebooks:  
  <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Cloud platforms with preconfigured environments:
  - See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) for Google Cloud.
  - Follow the [AWS Quickstart](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) for Amazon's Deep Learning AMI.
- Fully containerized training with [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>.

## Status

To check current CI status, click below:  
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>  
If the badge is green, all YOLOv5 Continuous Integration (CI) tests are passing ✅.

An Ultralytics engineer will assist you as soon as possible. Let us know if you have additional questions in the meantime! 😊

**Comment by rmarkovic00** [comment]
Thank you very much!

**Comment by sriram-dsl** [comment]
@rmarkovic00  When creating a custom dataset, there is no need to align your custom classes with COCO classes.  

You can define your own class names and structure, such as:  
```yaml
nc: 3  
names: [person, car, truck]  # or [car, person, truck] or [truck, person, car] – the order doesn't matter  
```  
However, you must ensure that the labeling data is correctly aligned with the class names in your dataset.  

For example, if your labels are:  
```
0: person  
1: car  
2: truck  
```
Then, the `names` should be:  
```yaml
names: [person, car, truck]
```  
Any CNN or ML model does not directly deal with class names—it only uses them for one-hot encoding and for human readability.
